401code_challenges/binarySearch/src/main/java/binarySearch/Library.java

```diff
@@ -0,0 +1,40 @@
+/*
+ * This Java source file was generated by the Gradle 'init' task.
+ */
+package binarySearch;
+
+public class Library {
+    public int binarySearch(int[] arr,int key)
+    {
+
+        //declare two indexes which are left and right
+        int leftIndex=0;
+        int rightIndedx =arr.length - 1;
```

Comment:
You definitely want to make sure you variable names are typo free! If you ever need to go back and add code to an existing code base and variables names have small typo it'll create a bug that's difficult to spot.